lib/mirage.ml

```diff
@@ -33,6 +33,83 @@ let driver_error name =
   Printf.sprintf "fail (Failure %S)" name
 
 (** {2 Devices} *)
+type qrexec = job
+let qrexec = job
+
+let qrexec_qubes = impl @@ object
+  inherit base_configurable
+  method ty = qrexec
+  val name = Name.ocamlify @@ "qrexec_"
+  method name = name
+  method module_name = "Qubes.RExec"
+  method packages = Key.pure ["mirage-qubes"]
+  method libraries = Key.pure ["mirage-qubes"]
+  method connect i modname _args =
+  match Key.(get (Info.context i) target) with
+  | `Unix | `MacOSX | `Xen ->
+  Fmt.strf
+  "return (`Ok ())"
+  | `Qubes ->
+     Fmt.strf
+"@[<v 2>\
+         %s.connect ~domid:0 () >>= fun qrexec ->@ \
+         Lwt.async (fun () ->@ \
```

Comment:
Not sure about this (disconnecting rexec whenever we receive a shutdown request).
Seems more like something the main unikernel application should decide about
(qubes-mirage-skeleton just happens to respond to rexec finishing by exiting).